src/evaluators/ForInStatement.js

```diff
@@ -142,7 +142,13 @@ function emitResidualLoopIfSafe(
       strictCode,
       blockEnv
     );
-    if (result instanceof Value && gen.empty() && modifiedBindings.size === 0 && modifiedProperties.size === 1) {
+    if (
+      result instanceof NormalCompletion &&
```

Comment:
Come to think of it, why not introduce a new subtype of NormalCompletion called "SimpleNormalCompletion"?